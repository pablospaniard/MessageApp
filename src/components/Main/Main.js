import React, { Component } from 'react'
import styled from 'styled-components'
import { Switch, Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'

import { All, Received, Sent } from '../Messages'

import SendContainer from '../../containers/SendContainer/SendContainer'

const StyledOutter = styled.div`
  background-color: ${props => props.theme.mainBgd};
  margin: 20px;
  height: 90vh;
  border: 1px solid ${props => props.theme.borderMain};
  overflow: scroll;
`

class Main extends Component {
  render() {
    const { messages, loading } = this.props
    return (
      <StyledOutter>
        <Switch>
          <Route
            exact
            path="/"
            component={() => <All messages={messages} loading={loading} />}
          />
          <Route
            path="/received"
            component={() => <Received messages={messages} loading={loading} />}
          />
          <Route
            path="/sent"
            component={() => <Sent messages={messages} loading={loading} />}
          />
          <Route path="/send-sms" component={SendContainer} />
          <Redirect from="*" to="/" />
        </Switch>
      </StyledOutter>
    )
  }
}

Main.propTypes = {
  messages: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  loading: PropTypes.bool
}

export default Main
