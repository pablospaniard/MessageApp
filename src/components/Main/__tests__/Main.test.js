import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'

import Main from '../Main'

const mockStore = configureMockStore()
const store = mockStore({})

const messages = {
  items: [
    {
      id: 'e8077d803532c0b5937c639b60216938',
      href:
        'https://rest.messagebird.com/messages/e8077d803532c0b5937c639b60216938',
      direction: 'mt',
      type: 'sms',
      originator: 'MessageBird',
      body: 'The message to be sent',
      reference: 'the-client-reference',
      validity: null,
      gateway: 240,
      typeDetails: {},
      datacoding: 'plain',
      mclass: 1,
      scheduledDatetime: null,
      createdDatetime: '2016-04-29T09:42:26+00:00',
      recipients: {
        totalCount: 1,
        totalSentCount: 1,
        totalDeliveredCount: 0,
        totalDeliveryFailedCount: 0,
        items: [
          {
            recipient: 31612345678,
            status: 'sent',
            statusDatetime: '2016-04-29T09:42:26+00:00'
          }
        ]
      }
    }
  ]
}
const noMessages = {
  items: []
}

describe('Main should render properly', () => {
  afterEach(cleanup)

  function renderWithRouter(
    ui,
    {
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] })
    } = {}
  ) {
    return {
      ...render(<Router history={history}>{ui}</Router>),
      history
    }
  }

  it('should match snapshot', () => {
    const { container } = renderWithRouter(<Main messages={messages} />)
    expect(container.firstChild).toMatchSnapshot('Main_snapshot_1')
  })

  it('should render a received page if pathname is received', () => {
    const { getByTestId } = renderWithRouter(
      <Provider store={store}>
        <Main messages={messages} />
      </Provider>,
      {
        route: '/received'
      }
    )
    expect(getByTestId('received'))
  })

  it('should render a sent page if pathname is sent', () => {
    const { getByTestId } = renderWithRouter(
      <Provider store={store}>
        <Main messages={messages} />
      </Provider>,
      {
        route: '/sent'
      }
    )
    expect(getByTestId('sent'))
  })

  it('should render a NoMessages page if no messages', () => {
    const { getByText } = renderWithRouter(
      <Provider store={store}>
        <Main messages={noMessages} />
      </Provider>,
      {
        route: '/'
      }
    )
    expect(getByText('You have not sent or received any messages.'))
  })
})
