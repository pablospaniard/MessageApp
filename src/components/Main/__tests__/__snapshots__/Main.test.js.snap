// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main_snapshot_1 1`] = `
<div
  class="sc-jTzLTM enCwFM"
>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Type
        </th>
        <th>
          Recipient
        </th>
        <th>
          Originator
        </th>
        <th>
          Message
        </th>
        <th>
          Status
        </th>
        <th>
          Date
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="sc-EHOje bdudsa"
        >
          Outgoing
        </td>
        <td
          class="sc-EHOje bdudsa"
        >
          31612345678
        </td>
        <td
          class="sc-EHOje bdudsa"
        >
          MessageBird
        </td>
        <td
          class="sc-EHOje bdudsa"
        >
          The message to be sent
        </td>
        <td
          class="sc-EHOje bdudsa"
        >
          sent
        </td>
        <td
          class="sc-EHOje bdudsa"
        >
          29-04-16 11:42
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
